File: code-samples/messaging/message-store.cs
using System;
using Newtonsoft.Json;
using System.Threading.Tasks;
using RingCentral;

namespace Read_MessageStore
{
    class Program
    {
	static void Main(string[] args)
	{
            restClient = new RestClient(
		Environment.GetEnvironmentVariable("RC_CLIENT_ID"),
		Environment.GetEnvironmentVariable("RC_CLIENT_SECRET"),
		Environment.GetEnvironmentVariable("RC_SERVER_URL"));
            restClient.Authorize(
		Environment.GetEnvironmentVariable("RC_USERNAME"),
		Environment.GetEnvironmentVariable("RC_EXTENSION"),
		Environment.GetEnvironmentVariable("RC_PASSWORD")).Wait();
	    read_user_message_store().Wait();
	}
	static private async Task read_user_message_store()
	{
	    var parameters = new ListMessagesParameters();
	    parameters.messageType = string[] ("SMS");
	    var response = await restClient.Restapi().Account().Extension().MessageStore().List(parameters);
	    
	    var jsonStr = JsonConvert.SerializeObject(response);
	    Console.WriteLine(jsonStr);
	}
    }
}

File: code-samples/messaging/message-store.java
import com.ringcentral.*;
import com.ringcentral.definitions.*;

class Read_MessageStore {
	public static void main(String[] args) {
		try {
			read_user_message_store();
		} catch (RestException | IOException e) {
			e.printStackTrace();
		}
	}

	public static void read_user_message_store() throws RestException, IOException {
		RestClient rc = new RestClient("client_id", "client_secret", "server_url");
		rc.authorize("username", "extension_number", "password");

		ListMessagesParameters parameters = new ListMessagesParameters();
		parameters.messageType = new String[] {"SMS"};

		var response = rc.restapi().account().extension().messagestore().list(parameters);

		String jsonStr = JSON.toJSONString(response);
		System.out.println(jsonStr);
	}
}
